stan/math/prim/mat/fun/trace_gen_quad_form.hpp

```diff
@@ -10,10 +11,11 @@ namespace math {
 /**
  * Compute trace(D B^T A B).
  **/
-template <int RD, int CD, int RA, int CA, int RB, int CB>
-inline double trace_gen_quad_form(const Eigen::Matrix<double, RD, CD> &D,
-                                  const Eigen::Matrix<double, RA, CA> &A,
-                                  const Eigen::Matrix<double, RB, CB> &B) {
+template <typename TD, int RD, int CD, typename TA, int RA, int CA, typename TB,
+          int RB, int CB, typename = enable_if_any_not_var<TD, TA, TB>>
+inline return_type_t<TD, TA, TB> trace_gen_quad_form(
+    const Eigen::Matrix<TD, RD, CD> &D, const Eigen::Matrix<TA, RA, CA> &A,
```

Comment:
Can you hit enter after D, putting function inputs on the same line?

Reply:
Yep!